### DISCUSSION TITLE
Dispose the controller when navigating to same page?

### DISCUSSION THREAD

**Discussion by bastaware** [jonataslaw/getx#2613] [Q&A]
I've got a search-field in the appbar of my application. It's available on every page of the app. When something is selected in the seach-field, it routes to a named page that loads and displays details about the entity. It is also possible to use the search bar from the details page, which means, that it routes to the same page that it's already on. As the id of the entity is part of the url, the route is actually different and as such everything works. But...

When the page is opened, my GetX controller does some initialization, and on leave it disposes of resources such as TextController.

My problem is that when I navigate to the same page, onClose is not called before leaving which means that the resources are not disposed. Rather it open the 'new' page, calls onInit() then calls onClose() and disposes the resources that are in use.

When the action of the seach-field determines that the currentRoute is already on the details-page, it uses `Get.offNamed('/details', parameters({id: entityId}))`, if it is on another page it simply uses `Get.toNamed('/details', parameters({id: entityId}))`

I've made a little app the illustrates the problem. The search-field has been replaced by buttons for simplicity.

[https://flutlab.io/editor/994c9029-47e9-46e2-a0e6-cfd970b82079](https://flutlab.io/editor/994c9029-47e9-46e2-a0e6-cfd970b82079)

To make it fail, try pressing 'To subpage' button twice.

**Notice in the console:**
Initialized
Initialized
closed

**I was expecting:**
Initialized
closed
Initialized
closed

**Comment by jaxnz** [comment]
@bastaware did you manage to find a solution to this?

**Comment by WMOH-DEV** [comment]
Getx doesnt support different data on the same page by navigation 
there're more than one solution but all of them has issues 
and i don't know why Getx till now has such this issue.

1- Dont navigate, but reload with the different data. ( no navigation and back button )
2- use Tag when you put/lazyput the controller so you can create multi instance of the same controller. (previous page replaced automatically so back button won't back to the right page)

**Comment by bayram31051993** [comment]
delete your controller
Get.delete<YourController>();
Get.to(()=>YourSamePage());
